MHVLib/Connection/MHVConnection.m

```diff
@@ -257,7 +257,21 @@ - (void)executeRestRequest:(MHVHttpServiceRequest *)request
             return;
         }
         
-        if (error)
+        if (response.hasError)
```

Comment:
1. It should be fine to add the responseAsString to the error. 2. It probably would be a good idea to create a 'MHVNetworkError' or some other error type that is more specific to making a request to the HealthVault service.